### PR TITLE
Extract DevTools code from src/planning

### PR DESCRIPTION
**Pull request by piotrswigon**
The main objective is to separate planning code from the concrete implementation of Arcs Explorer. Similar work has been done for the runtime-proper code in #3048.

This is a a next step in the plan outlined below:
https://docs.google.com/document/d/1DC8gDV-lzwYkh60kMj661eT-clNLircMa0YGw1whgI8/edit
(google only, sorry!)

What happens here concretely:
* PlannerInspector and its implementation, DevtoolsPlannerInspector are introduced.
* planning-explorer-adapter.ts and strategy-explorer-adapter.ts are subsumed by PlannerInspector.
* I've removed support for strategy explorer from the recipe-index, I don't think anyone is using it.
* From now, one needs to explicitly pass DevtoolsPlannerInspectorFactory to planificator or planner. I've done it for PlanningTestHelper, but not everywhere. Will do on per-test basis when needed.

@mmandlis , can you comment whether the dead code (debug methods) in StrategyExplorerAdapter are still needed? If so, where you would like me to put them?

### CODE REVIEW

**Comment by piotrswigon** [comment]
Also 2745.7 -> 2742.1 = 3.6 automated points for various improvements along the way.

File: shells/pipes-shell-2/lib/utils.js [hunk 1]
```diff
@@ -38,7 +39,9 @@ export const createPlanificator = async arc => {
     userid: 'user',
     storageKeyBase: 'volatile',
     onlyConsumer: false,
-    debug: true
+    //kill with fire
```
Comment (mmandlis) [inline shells/pipes-shell-2/lib/utils.js]: is the comment intentional? 
i see a similar one below, so probably yes - can you rephrase it to be both more friendly and informative, please :)

File: src/devtools-connector/devtools-planner-inspector.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright (c) 2018 Google Inc. All rights reserved.
```
Comment (mmandlis) [inline src/devtools-connector/devtools-planner-inspector.ts]: nit - 2019?

File: shells/pipes-shell-2/lib/utils.js (hunk 1 above)
Comment (piotrswigon) [reply shells/pipes-shell-2/lib/utils.js]: Haha, no, it's a marker to revisit before sending out. This obviously failed :-D will correct when at the computer.

Comment (piotrswigon) [reply shells/pipes-shell-2/lib/utils.js]: Removed the comment now, I hoped to remove the 'debug' flag, but didn't manage to do it yet.

File: src/devtools-connector/devtools-planner-inspector.ts (hunk 2 above)
Comment (piotrswigon) [reply src/devtools-connector/devtools-planner-inspector.ts]: Done :) Time flies by!
